packages/perseus/src/widgets/interactive-graphs/locked-line.tsx

```diff
@@ -15,52 +16,72 @@ type Props = LockedLineType & {
 };
 
 const LockedLine = (props: Props) => {
-    const {
-        color,
-        lineStyle,
-        kind,
-        points,
-        showStartPoint,
-        showEndPoint,
-        range,
-    } = props;
+    const {color, lineStyle, kind, points, showArrows, range} = props;
     const [point1, point2] = points;
+    // NOTE: Segments should not show both a point and an arrow.
+    //       Arrows supersede points.
+    const showStartPoint =
+        props.showStartPoint && (kind !== "segment" || !showArrows);
+    const showEndPoint =
+        props.showEndPoint && (kind !== "segment" || !showArrows);
+
+    let arrowHeadValues = getArrowheadValues(point1.coord, point2.coord, range);
+    let arrowTip = kind === "segment" ? point2.coord : arrowHeadValues.tip;
```

Comment:
The fact that we need this ternary to ignore `arrowHeadValues.tip` in the case of a segment graph makes me think that `getArrowheadValues` is the wrong abstraction. If it were inlined, this code could become:

```ts
const direction = vec.sub(throughPoint, referencePoint);
const angle = calculateAngleInDegrees(direction);
const tip = kind === "segment"
    ? throughPoint
    : getIntersectionOfRayWithBox(throughPoint, referencePoint, range);
```

Reply:
Updated.